data/transactions/logic/assembler.go

```diff
@@ -1335,7 +1335,7 @@ func typeDupTwo(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, e
 func typeSelect(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes, error) {
 	top := len(pgm.stack) - 1
 	if top >= 2 {
-		if pgm.stack[top-1] == pgm.stack[top-2] {
+		if pgm.stack[top-1].AVMType == pgm.stack[top-2].AVMType {
```

Comment:
is the union of static(32) and static(8) some new construct like one-of(static(8), static(32),...)
or is it just {8,32}

Reply:
Worth pointing out, by the way, that if `union` is sufficiently smart, many of these `type*` functions get simplified.  if `union(bytes,uint) == any`, then there's no need to check if the `AVMType` of `top-1` and `top-2` types are equal - just return the `union`.

Reply:
For Bobs example, it'd currently fail the `Assignable` check since there is no overlap in types. 

For union'ing bytes/uint, when would that occur? Shouldnt the stack types be either bytes xor uint (or any already)?

Reply:
Its possible you could reach a statement about the type of C (talking about `a;b;c;select`) such that it could never be 0, so you can pass along the return as always the B type, which is an improvement over `any`.  I suppose this is true of the index of `LOADS/STORES`

Reply:
yes, that's certainly true and would happen before the call to `union`, so it was too strong for me to say you _always_ return the union.